modules/core/src/main/java/org/apache/ignite/internal/pagemem/wal/record/delta/PurgeRecord.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.pagemem.wal.record.delta;
+
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.pagemem.PageMemory;
+import org.apache.ignite.internal.processors.cache.persistence.tree.io.BPlusIO;
+import org.apache.ignite.internal.processors.cache.persistence.tree.io.PageIO;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ * Removal of multiple items (designated by their 0-based indexes) in an index leaf page.
+ */
+public class PurgeRecord extends PageDeltaRecord {
+    /** Indexes of items to remove from the page. */
+    private int[] items;
+
+    /** Number of used elements in items array. */
+    private int itemsCnt;
+
+    /** Resulting count of items that should remain on the page. */
+    private int cnt;
```

Comment:
For now, we use `cnt` only for assertion.
I think we shouldn't store some data on the disk if it used only for assertion that can be disabled in production environment.

Please, clarify, why we need this check

```
assert cnt0 == cnt + itemsCnt : "unexpected count: cnt0=" + cnt0 + ", cnt=" + cnt + ", itemsCnt=" + itemsCnt;
```

If we really need this check it should not be `assert` but regular `if` statement with the throw if true .